src/jaxsim/api/common.py

```diff
@@ -44,7 +43,7 @@ class ModelDataWithVelocityRepresentation(JaxsimDataclass, abc.ABC):
     @contextlib.contextmanager
     def switch_velocity_representation(
         self, velocity_representation: VelRepr
-    ) -> ContextManager[Self]:
+    ) -> contextlib.AbstractContextManager[Self]:
```

Comment:
Mmh this has always been tricky. In practice, this method is a generator that yields an object from another generator. I never managed to get the right configuration that makes the code analysis of PyCharm happy. Is `pyupgrade` happy if the returned type is `Iterator[Self]`?

Reply:
It should be happy with that as it's not a type checker

Reply:
Addressed in https://github.com/ami-iit/jaxsim/pull/226/commits/2ae3689f2cd7f9616d3b850e5c323c81acc9d9ee